libuavcan/src/transport/uc_transfer_buffer.cpp

```diff
@@ -96,18 +96,43 @@ void TransferBufferManagerEntry::Block::destroy(Block*& obj, IPoolAllocator& all
     }
 }
 
+// precondition is that target_offset < total_offset + Block::Size
 void TransferBufferManagerEntry::Block::read(uint8_t*& outptr, unsigned target_offset,
                                              unsigned& total_offset, unsigned& left_to_read)
 {
     UAVCAN_ASSERT(outptr);
-    for (unsigned i = 0; (i < Block::Size) && (left_to_read > 0); i++, total_offset++)
+    // total offset == offset at the start of this block
+    const unsigned start_of_block = total_offset;
+    const unsigned end_of_block = total_offset + Block::Size;
+
+    unsigned bytes_to_copy = 0;
+
+    // Case 1 -- start of wanted area is in this Block
+    if (target_offset >= start_of_block && target_offset < end_of_block)
     {
-        if (total_offset >= target_offset)
+        const unsigned bytes_available_at_offset = end_of_block - target_offset;
+        bytes_to_copy = bytes_available_at_offset;
+        if (bytes_to_copy > left_to_read)
         {
-            *outptr++ = data[i];
-            left_to_read--;
+            bytes_to_copy = left_to_read;
         }
+        uint8_t *src = data + (target_offset - start_of_block);
```

Comment:
`uint8_t *src` :scream_cat:

Reply:
What about the coding style violations though. I now understand that my previous comment might be overly cryptic, but what I was hinting at is that the asterisk is misplaced. I recall you made a config file for Clang autoformatter, maybe it is time to use it here?